src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2384,21 +2384,37 @@ private bool ConsumeCommentMultiSegment()
                 ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.InvalidCharacterAtStartOfComment, marker);
             }
 
+            bool allowEmptyComment = marker == JsonConstants.Slash;
+
             localBuffer = localBuffer.Slice(1);
             if (localBuffer.Length == 0)
             {
-                if (IsLastSpan)
-                {
-                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWhileReadingComment);
-                }
-                if (!GetNextSpan())
+                if (IsLastSpan || !GetNextSpan())
                 {
                     if (IsLastSpan)
                     {
-                        ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWhileReadingComment);
+                        if (allowEmptyComment)
+                        {
+                            if (_tokenType != JsonTokenType.Comment)
+                            {
+                                _previousTokenType = _tokenType;
+                            }
+
+                            _totalConsumed++;
+                            _bytePositionInLine++;
+                            _tokenType = JsonTokenType.Comment;
+                            _consumed += 2;
```

Comment:
Hmm... I don't think this is correct.

If you check `BytesConsumed`, this ends up setting it incorrectly:
```C#
[Theory]
[InlineData("//", "", 1, 2)] // expected - 2 / actual - 4
[InlineData("//", "", 2, 2)] // expected - 2 / actual - 3
[InlineData("//a", "a", 2, 3)] // expected - 3 / actual - 2
[InlineData("//abc", "abc", 2, 5)] // expected - 5 / actual - 4
public static void JsonWithSingleLineCommentWithNoLineEndingsFinalBlockMultiSegment(string jsonString, string expectedComment, int segmentSize, int consumed)
{
    byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
    var state = new JsonReaderState(options: new JsonReaderOptions { CommentHandling = JsonCommentHandling.Allow });
    ReadOnlySequence<byte> sequence = JsonTestHelper.GetSequence(dataUtf8, segmentSize);
    var json = new Utf8JsonReader(sequence, isFinalBlock: true, state);

    Assert.True(json.Read());
    Assert.Equal(JsonTokenType.Comment, json.TokenType);
    Assert.Equal(expectedComment, json.GetComment());
    Assert.Equal(consumed, json.BytesConsumed); // this fails for the 4 test cases listed
    Assert.False(json.Read());
}
```

Reply:
will take a look - thanks!